### PR TITLE
Upgrade to core v6.0.25 and sync v5.0.22

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+x.x.x Release notes (yyyy-MM-dd)
+=============================================================
+NOTE: This version bumps the Realm file format to version 11. It is not possible to downgrade version 10 or earlier. Moreover, older files will automatically be upgraded to the new file format. Files created by Realm JavaScript prior to v1.0.0, might not be upgradeable. 
+
+### Enhancements
+* None.
+
+### Fixed
+* Upgrading files with string primary keys would result in a file where it was not possible to find the objects by primary key. ([realm/realm-cocoa#6716](https://github.com/realm/realm-cocoa/issues/6716), since v6.0.4)
```
Comment (tgoyne) [inline CHANGELOG.md:9]: This is a core version number and not a JS version number.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+
+### Compatibility
+* Realm Object Server: 3.23.1 or later.
+* Realm Studio: 3.11 or later.
```
Comment (tgoyne) [inline CHANGELOG.md:13]: Realm studio 3.11 can't open v11 files.

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [reply CHANGELOG.md:9]: The core version is v6.0.7 (https://github.com/realm/realm-core/pull/3893/) while v6.0.4 refers to the corresponding JS version. Let me it a bit clearer.

File: CHANGELOG.md (hunk 2 above)
Comment (kneth) [reply CHANGELOG.md:13]: No, Studio hasn't been updated yet and it should be clear that no Studio version can open this format.

**Comment by kneth** [comment]
@tgoyne Thanks for reviewing!

**Pull request merged**